src/optimization/problem.cpp

```diff
@@ -252,41 +253,58 @@ ExitStatus Problem::solve(const Options& options, [[maybe_unused]] bool spy) {
     }
 #endif
 
-    // Invoke interior-point method solver
-    status = interior_point(
-        InteriorPointMatrixCallbacks{
-            [&](const Eigen::VectorXd& x) -> double {
-              x_ad.set_value(x);
-              return f.value();
-            },
-            [&](const Eigen::VectorXd& x) -> Eigen::SparseVector<double> {
-              x_ad.set_value(x);
-              return g.value();
-            },
-            [&](const Eigen::VectorXd& x, const Eigen::VectorXd& y,
-                const Eigen::VectorXd& z) -> Eigen::SparseMatrix<double> {
-              x_ad.set_value(x);
-              y_ad.set_value(y);
-              z_ad.set_value(z);
-              return H.value();
-            },
-            [&](const Eigen::VectorXd& x) -> Eigen::VectorXd {
-              x_ad.set_value(x);
-              return c_e_ad.value();
-            },
-            [&](const Eigen::VectorXd& x) -> Eigen::SparseMatrix<double> {
-              x_ad.set_value(x);
-              return A_e.value();
-            },
-            [&](const Eigen::VectorXd& x) -> Eigen::VectorXd {
-              x_ad.set_value(x);
-              return c_i_ad.value();
-            },
-            [&](const Eigen::VectorXd& x) -> Eigen::SparseMatrix<double> {
-              x_ad.set_value(x);
-              return A_i.value();
-            }},
-        m_iteration_callbacks, options, x);
+    auto [bound_constraint_mask, bounds, conflicting_bound_indices] =
+        get_bounds(m_decision_variables, m_inequality_constraints, A_i.value());
+    if (!conflicting_bound_indices.empty()) {
+      status = ExitStatus::GLOBALLY_INFEASIBLE;
```

Comment:
You can do `return ExitStatus::GLOBALLY_INFEASIBLE;` in this if statement to avoid the extra if nesting later.

Reply:
Early-returning doesn't print the exit status; I did try moving the exit status print to a `scope_exit` but I felt this made the control flow non-obvious at the time. In retrospect, that print is fairly minor so maybe it's just not a big deal to do that.

Reply:
Oh jk, I didn't notice that this changed after I rebased even though I had to manually resolve the conflict :facepalm: